crates/python/src/qpu/isa.rs

```diff
@@ -0,0 +1,184 @@
+use qcs_api_client_openapi::models::{
+    Architecture1, Characteristic, Edge, Family, InstructionSetArchitecture, Node, Operation,
+    OperationSite, Parameter,
+};
+use rigetti_pyo3::{
+    create_init_submodule, py_wrap_data_struct, py_wrap_error, py_wrap_simple_enum,
+    pyo3::{
+        exceptions::{PyRuntimeError, PyValueError},
+        prelude::*,
+        types::{PyFloat, PyInt, PyList, PyString},
+        Py, Python,
+    },
+    wrap_error, ToPython, ToPythonError,
+};
+
+use qcs::qpu::get_isa;
+
+use crate::qpu::client::PyQcsClient;
+
+create_init_submodule! {
+    classes: [
+        PyFamily,
+        PyEdge,
+        PyNode,
+        PyArchitecture1,
+        PyCharacteristic,
+        PyParameter,
+        PyOperationSite,
+        PyOperation,
+        PyInstructionSetArchitecture
+    ],
+    errors: [
+        QcsIsaSerializationError,
+        QcsIsaError
+    ],
+    funcs: [
+        py_get_isa
+    ],
+}
+
+wrap_error!(IsaSerializationError(serde_json::Error));
```

Comment:
Wow, this all looks like it took some patience!

Reply:
Definitely glad to have these macros though, and it would probably be difficult to have a clean interface around something that wraps nested structs.